packages/network-console-shared/src/file_io/postman/v2.1/url.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+    URLObject,
+    VariableType,
+} from '../../../collections/postman/v2.1/schema-generated';
+import { INetConsoleParameter } from '../../../net/net-console-http';
+
+export function constructURLObject(url: string, routeParams: INetConsoleParameter[], queryParameters: INetConsoleParameter[]) {

```

Comment:
It would be nice to add some comments here in jsdoc to explain why we need our own parser/handler for URLs. People new to this code might wonder why we didn't use the `URL` API.
So, mentioning here that we need it so we can parse things like `{{uriroot}}` parameters and such would help.